kernel-install/51-mkosi-addon.install

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import argparse
+import logging
+import os
+import sys
+from pathlib import Path
+
+from mkosi.initrd import KernelInstallContext
+from mkosi.log import die, log_setup
+from mkosi.qemu import KernelType
+from mkosi.run import run, uncaught_exception_handler
+from mkosi.types import PathString
+from mkosi.util import mandatory_variable
+
+
+@uncaught_exception_handler()
+def main() -> None:
+    log_setup()
+
+    # No local configuration? Then nothing to do
+    if not Path("/etc/mkosi-addon").exists() and not Path("/run/mkosi-addon").exists():
+        return
+
+    parser = argparse.ArgumentParser(
+        description="kernel-install plugin to build local addon for initrd/cmdline/ucode",
+        allow_abbrev=False,
+        usage="51-mkosi-addon.install COMMAND KERNEL_VERSION ENTRY_DIR KERNEL_IMAGE…",
+    )
+
+    KernelInstallContext.kernel_install_arg_parse(parser)
+
+    context = KernelInstallContext(
+        **vars(parser.parse_args()),
+        staging_area=Path(mandatory_variable("KERNEL_INSTALL_STAGING_AREA")),
+        layout=mandatory_variable("KERNEL_INSTALL_LAYOUT"),
+        image_type=mandatory_variable("KERNEL_INSTALL_IMAGE_TYPE"),
+        verbose=int(os.getenv("KERNEL_INSTALL_VERBOSE", 0)) > 0,
+    )
+
+    if context.command != "add" or not context.layout == "uki":
+        return
+
+    if not context.kernel_image or not context.kernel_image.exists():
+        die(f"Kernel image {context.kernel_image} does not exist")
```

Comment:
Please skip in this case as well, let's leave validation of the kernel image argument to `kernel-install`